star-dash/star-dash/GameEngine/Systems/Complex/CollisionSystem.swift

```diff
@@ -45,7 +45,14 @@ class CollisionSystem: System {
     }
 
     private func handlePlayerMonsterContactEvent(event: PlayerMonsterContactEvent) {
```

Comment:
Do you think it makes sense to shift this logic under `MonsterSystem`? Since this is all monster related logic, I feel that its ideal we eventually remove `CollisionSystem` completely.